site/e2e/reporter.ts

```diff
@@ -1,85 +1,145 @@
   TestResult,
   FullResult,
   Reporter,
+  TestError,
 } from "@playwright/test/reporter";
 import axios from "axios";
+import type { Writable } from "stream";
 
 class CoderReporter implements Reporter {
+  config: FullConfig | null = null;
+  testOutput = new Map<string, Array<[Writable, string]>>();
+  passedCount = 0;
+  failedTests: TestCase[] = [];
+  timedOutTests: TestCase[] = [];
+
   onBegin(config: FullConfig, suite: Suite) {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(`Starting the run with ${suite.allTests().length} tests`);
+    this.config = config;
+    console.log(`==> Running ${suite.allTests().length} tests`);
   }
 
   onTestBegin(test: TestCase) {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(`Starting test ${test.title}`);
+    this.testOutput.set(test.id, []);
+    console.log(`==> Starting test ${test.title}`);
   }
 
-  onStdOut(chunk: string, test: TestCase, _: TestResult): void {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(
-      `[stdout] [${test ? test.title : "unknown"}]: ${chunk.replace(
-        /\n$/g,
-        "",
-      )}`,
-    );
+  onStdOut(chunk: string, test?: TestCase, _?: TestResult): void {
+    if (!test) {
+      const preserve = this.config?.preserveOutput === "always";
+      if (preserve) {
+        console.log(`[stdout] ${chunk.replace(/\n$/g, "")}`);
+      }
+      return;
+    }
+    this.testOutput.get(test.id)!.push([process.stdout, chunk]);
   }
 
-  onStdErr(chunk: string, test: TestCase, _: TestResult): void {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(
-      `[stderr] [${test ? test.title : "unknown"}]: ${chunk.replace(
-        /\n$/g,
-        "",
-      )}`,
-    );
+  onStdErr(chunk: string, test?: TestCase, _?: TestResult): void {
+    if (!test) {
+      const preserve = this.config?.preserveOutput === "always";
+      if (preserve) {
+        console.error(`[stderr] ${chunk.replace(/\n$/g, "")}`);
+      }
+      return;
+    }
+    this.testOutput.get(test.id)!.push([process.stderr, chunk]);
   }
 
   async onTestEnd(test: TestCase, result: TestResult) {
-    // eslint-disable-next-line no-console -- Helpful for debugging
-    console.log(`Finished test ${test.title}: ${result.status}`);
+    console.log(`==> Finished test ${test.title}: ${result.status}`);
+
+    if (result.status === "passed") {
+      this.passedCount++;
+    }
+
+    if (result.status === "failed") {
+      this.failedTests.push(test);
+    }
 
-    if (result.status !== "passed") {
-      // eslint-disable-next-line no-console -- Helpful for debugging
-      console.log("errors", result.errors, "attachments", result.attachments);
+    if (result.status === "timedOut") {
+      this.timedOutTests.push(test);
     }
-    await exportDebugPprof(test.title);
+
+    const outputFile = `test-results/debug-pprof-goroutine-${test.title}.txt`;
```

Comment:
nit: It might be easier to interact in shell with files without spaces, so: `s/ /-/g` (possible improvement)